tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_rand_uint.h

```diff
@@ -0,0 +1,30 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+template <bool APPROXIMATION_MODE>
+inline void rand_uint_init(uint seed) {
+    init_prng_seed(seed);
+}
+
+template <bool APPROXIMATION_MODE>
+inline void rand_uint() {
+#pragma GCC unroll 0
+    for (int d = 0; d < 8; d++) {
+        TTI_SFPMOV(0, 9, p_sfpu::LREG0, 8);
+        TTI_SFPSETSGN(0, p_sfpu::LREG0, p_sfpu::LREG0, 1);
```

Comment:
Can you explain why the SFPSETSGN instruction is needed? Is this related to your comments in the writer kernel "The hardware PRNG is not uniformly distribute."

Reply:
The sign bit is just the MSB, and when working with uint32 numbers it isn't used as a sign bit, it's actually used as 2^31. So by zeroing it you would limit the range to be half of uint32 max value.

Reply:
@BuiChiTrung without the `TTI_SFPSETSGN` instruction, were you seeing negative output from the prng? Or did you use `TTI_SFPSETSGN` as just in case?

Reply:
Just checked again. I saw negative numbers in first place because I mistakenly used Int32 CB to get result from sfpu. Updated [here](https://github.com/tenstorrent/tt-metal/pull/13735/commits/992a2c0c2af023fd81e1a82b35c8ae4a1258e70c).